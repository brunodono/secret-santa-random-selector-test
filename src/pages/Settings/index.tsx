import Card from "components/Card"
import Footer from "components/Footer"
import Form from "components/Form"
import ParticipantsList from "components/ParticipantsList"

const Settings = () => {
    return (
        <>
            <Card>
                <section>
                    <h2>Let's get started!</h2>
                    <Form />
                    <ParticipantsList />
                    <Footer />
                </section>
            </Card>

        </>

    )
}

export default Settings