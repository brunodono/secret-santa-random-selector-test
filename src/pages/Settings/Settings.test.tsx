import { render } from "@testing-library/react";
import React from "react";
import { RecoilRoot } from "recoil";
import Settings from ".";

const mockBrowsing = jest.fn();

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockBrowsing
    }
});

describe('the settings page',() => {
    test('has to render correctly', () => {
        const { container } = render(
        <RecoilRoot>
            <Settings />
        </RecoilRoot>)

        expect(container).toMatchSnapshot()
    })
})