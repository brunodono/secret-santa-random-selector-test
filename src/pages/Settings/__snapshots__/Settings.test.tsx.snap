// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the settings page has to render correctly 1`] = `
<div>
  <form>
    <input
      placeholder="Enter the participants's names"
      type="text"
      value=""
    />
    <button
      disabled=""
    >
      Add
    </button>
    
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Start Game
    </button>
  </footer>
</div>
`;
